internal/controller/profiles/dev/states_dev.go

```diff
@@ -111,6 +112,14 @@ func (e *ensureRunningWorkflowState) Do(ctx context.Context, workflow *operatora
 	}
 	objs = append(objs, service)
 
+	if monitoring.IsMonitoringEnabled(pl) {
```

Comment:
Can you isolate in a function to avoid conditionals in this func? Otherwise, it will grow in complexity.